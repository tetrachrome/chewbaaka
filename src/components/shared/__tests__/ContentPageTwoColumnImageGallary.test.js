/**
 * ContentPageTwoColumnImageGallary.test.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 26, 2020
 * Updated  : Jul 26, 2020
 */

import React from 'react'

import { render } from '@testing-library/react'

import renderer from 'react-test-renderer'

import ContentPageTwoColumnImageGallary from '../ContentPageTwoColumnImageGallary'

const parts = [
  {
    image: "some image 1",
    caption: "This is image 1.",
    width: 720,
    height: 480
  },
  {
    image: "some image 2",
    caption: "This is image 2.",
    credit: "Image God",
    width: 1080,
    height: 720
  }
];

test('renders ContentPageTwoColumnImageGallary component', () => {
  render(
    <ContentPageTwoColumnImageGallary
      parts={parts}
    />
  );
});

test('ContentPageTwoColumnImageGallary component snapshot', () => {
  const tree = renderer
    .create(
      <ContentPageTwoColumnImageGallary
        parts={parts}
      />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
