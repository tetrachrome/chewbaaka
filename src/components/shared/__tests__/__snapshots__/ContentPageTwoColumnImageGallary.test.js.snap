// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentPageTwoColumnImageGallary component snapshot 1`] = `
<div
  className="ContentPageSubsectionTwoColumnContentTemplateOuterContainer"
>
  <div
    className="ContentPageSubsectionTwoColumnContentTemplateInnerContainer"
  >
    <div
      className="ContentPageSubsectionTwoColumnContentTemplateColumnContainer"
    >
      <div
        className="ContentPageSubsectionTwoColumnContentTemplateColumnInnerContainer ContentPageSubsectionTwoColumnContentTemplateColumnInnerContainerLhs"
      >
        <h4
          className="ContentPageSubsectionSubtitle"
        />
        <div
          className="ImageViewOuterContainer"
          style={
            Object {
              "maxWidth": 720,
            }
          }
        >
          <div
            className="ImageViewInnerContainer"
          >
            <div
              className="ImageViewCore"
            >
              <img
                alt="This is image 1."
                className="ImageViewImg"
                src="some image 1"
                style={
                  Object {
                    "width": 720,
                  }
                }
              />
              <div
                data-testid="ImageViewComponentCaptionPart"
              >
                <div
                  className="CaptionContainer"
                  data-testid="CaptionComponentTestId"
                  style={
                    Object {
                      "maxWidth": 720,
                    }
                  }
                >
                  <div
                    className="IconContainer"
                  >
                    <i
                      aria-hidden="true"
                      className="image icon"
                      onClick={[Function]}
                    />
                  </div>
                  <p
                    className="ContentPageCaptionTextSize CaptionText"
                  >
                    This is image 1.
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="ContentPageSubsectionTwoColumnContentTemplateColumnInnerContainer ContentPageSubsectionTwoColumnContentTemplateColumnInnerContainerRhs"
      >
        <h4
          className="ContentPageSubsectionSubtitle"
        />
        <div
          className="ImageViewOuterContainer"
          style={
            Object {
              "maxWidth": 1080,
            }
          }
        >
          <div
            className="ImageViewInnerContainer"
          >
            <div
              className="ImageViewCore"
            >
              <img
                alt="This is image 2. (Image credit: Image God)"
                className="ImageViewImg"
                src="some image 2"
                style={
                  Object {
                    "width": 1080,
                  }
                }
              />
              <div
                data-testid="ImageViewComponentCaptionPart"
              >
                <div
                  className="CaptionContainer"
                  data-testid="CaptionComponentTestId"
                  style={
                    Object {
                      "maxWidth": 1080,
                    }
                  }
                >
                  <div
                    className="IconContainer"
                  >
                    <i
                      aria-hidden="true"
                      className="image icon"
                      onClick={[Function]}
                    />
                  </div>
                  <p
                    className="ContentPageCaptionTextSize CaptionText"
                  >
                    This is image 2. (Image credit: Image God)
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
